rwviewer/views/TextViewer.cpp

```diff
@@ -0,0 +1,251 @@
+    if (textEdit->toPlainText().compare(newText)) {
+        textEdit->setText(newText);
+    }
+
+    updateRender();
+}
+
+void TextViewer::updateRender() {
+    viewerWidget->clearText();
+    const int ROW_STRIDE = currentFontSize * 1.2;
+    const int COL_STRIDE = currentFontSize * 1.2;
+    {
+        TextRenderer::TextInfo ti;
+        ti.font = currentFont;
+        ti.size = currentFontSize;
+        ti.baseColour = glm::u8vec3(255);
+        ti.backgroundColour = glm::u8vec4(0, 0, 0, 0);
+        ti.align = TextRenderer::TextInfo::TextAlignment::Left;
+        ti.wrapX = 0;
+
+        ti.text = currentGameString;
+        ti.screenPosition = glm::vec2(10, 10);
+        viewerWidget->showText(ti);
+    }
+
+    {
+        TextRenderer::TextInfo ti;
+        ti.font = currentFont;
+        ti.size = currentFontSize;
+        ti.baseColour = glm::u8vec3(255);
+        ti.backgroundColour = glm::u8vec4(0, 0, 0, 0);
+        ti.align = TextRenderer::TextInfo::TextAlignment::Left;
+        ti.wrapX = 0;
+
+        for(GameStringChar c=0x20; c<0xb2; ++c) {
+            unsigned column = c % 0x10;
+            unsigned row = (c / 0x10) - 2 + 3; /* +3 to offset first line*/
+            ti.text = c;
+            ti.screenPosition = glm::vec2(10 + (column * COL_STRIDE), 10 + (row * ROW_STRIDE));
+            viewerWidget->showText(ti);
+        }
+    }
+}
+
+void TextViewer::worldChanged() {
+    auto textNames = getFontTextureNames();
+    TextMapType textMap;
+    LoaderGXT loader;
+    std::set<std::string> keys;
+    for(const auto &textName : textNames) {
+        GameTexts texts;
+        auto handle = world()->data->index.openFile(textName);
+        loader.load(texts, handle);
+        const auto &language = textName;
+        textMap.languages.push_back(language);
+        const auto &stringTable = texts.getStringTable();
+        for (const auto &tableItem : stringTable) {
+            keys.insert(tableItem.first);
+            textMap.map_lang_key_tran[language][tableItem.first] = tableItem.second;
+        }
+    }
+    textMap.keys.resize(keys.size());
+    std::move(keys.begin(), keys.end(), textMap.keys.begin());
+
+    textModel->setData(textMap);
+}
+
+std::vector<std::string> TextViewer::getFontTextureNames() {
+    const auto &gameDataPath = rwfs::path(world()->data->getDataPath());
+    rwfs::path textPath;
+    for (const rwfs::path &p : rwfs::directory_iterator(gameDataPath)) {
+        if (!rwfs::is_directory(p)) {
+            continue;
+        }
+        std::string filename = p.filename().string();
+        std::transform(filename.begin(), filename.end(), filename.begin(), ::tolower);
+        if (!filename.compare("text")) {
+            textPath = p;
+            break;
+        }
+    }
+    if (!textPath.string().length()) {
+        throw std::runtime_error("text directory not found in gamedata path");
+    }
+    std::vector<std::string> names;
```

Comment:
It should be doable to use `reserve`.

Reply:
That vector just contains a list of all text localisations (american, english, french, ...) . Normally it is short so reserve should not be needed.